src/rider/main/kotlin/com/jetbrains/rider/plugins/efcore/features/migrations/add/AddMigrationVfsListenerService.kt

```diff
@@ -0,0 +1,46 @@
+package com.jetbrains.rider.plugins.efcore.features.migrations.add
+
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.vfs.isFile
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.events.VFileCreateEvent
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
+import com.jetbrains.rd.util.reactive.Signal
+import kotlinx.coroutines.CoroutineScope
+
+@Service(Service.Level.PROJECT)
+class AddMigrationVfsListenerService(intellijProject: Project, projectScope: CoroutineScope) {
```

Comment:
Well this specific listener is not related to Add migration, so let's rename it to something like
```suggestion
class FileEventsListenerService(intellijProject: Project, projectScope: CoroutineScope) {
```